server/lib/src/server/mod.rs

```diff
@@ -1500,8 +1513,33 @@ impl<'a> QueryServerWriteTransaction<'a> {
     pub(crate) fn reload_domain_info(&mut self) -> Result<(), OperationError> {
         let domain_name = self.get_db_domain_name()?;
         let display_name = self.get_db_domain_display_name()?;
+        let domain_ldap_allow_unix_pw_bind = match self.get_domain_ldap_allow_unix_pw_bind() {
```

Comment:
Here you're trying represent "the possibility of abscence" but you're encoding that into the Result type. Result shows "the possibility of error".

If you want to still do it this way, you want to change the type of get_domain_ldap_allow_unix_pw_bind to `Result<Option<bool>, OperationError>`. This way you show that there is a possibility of error, and if there was no error, that the contained value has a possibility of absecence. Then you con do "unwrap_or(true)" which allows you to set a default value here.

Reply:
But if I default the `get` method to return false on absence of an error then i wouldnt have to default anything in this relaod function right?

Reply:
Correct. I was more explaining there are two ways to approach it :)

Reply:
(Also you want to return true by default as allow unix pw bind is the default for now).